src/plugins/validate-semantic/validators/2and3/schemas.js

```diff
@@ -22,3 +22,40 @@ export const validate2And3TypeArrayRequiresItems = () => (system) => {
       }, [])
     })
 }
+
+export const validate2And3TypesInDefaultValuesMatchesWithEnum = () => (system) => {
+  return system.validateSelectors
+    .allSchemas()
+    .then(nodes => {
+      return nodes.reduce((acc, node) => {
+        const schemaObj = node.node
+        const { type } = schemaObj || {}
+        const enumeration = schemaObj.enum
+        var isValidFormat = true
+        var enumIndex = 0
+        enumeration.forEach((element, index) => {
+          if (type === "array" && (!Array.isArray(element) || element === null)) {
+            isValidFormat = false
+            enumIndex = index
+          } else if ((type === "number" || type === "string" || type === "boolean") && !(typeof element === type)) {
+            isValidFormat = false
+            enumIndex = index
+          } else if (type === "integer" && !Number.isInteger(element)) {
+            isValidFormat = false
+            enumIndex = index
+          } else if (type === "object" && ((element === null) || !(typeof element === type))) {
+            isValidFormat = false
+            enumIndex = index
+          }
+        })
+        if (!isValidFormat) {
+          acc.push({
+            message: "enum value should conform to its schema's `type`",
+            path: [...node.path, "enum", enumIndex],
+            level: "warning",
+          })
```

Comment:
Just a thought — unless I'm missing something here, it'd be simpler to push to our `acc` array while we're iterating over the enum items inside of `enumeration.forEach()`.